# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студенты гр. 2384 Лукьянова Мария и Пчелинцева Кристина.

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

**Интервальная оценка** - статистическая оценка, которая задается двумя числами (начало и конец интервала где может находится истинное значение параметра $\theta$)

**Точность $\theta^*$** - значение $\delta$ статистическая оценка параметра $\theta^*$, которое удовлетворяет неравенству:

$$|\theta^*-\theta|<\delta$$

**Наддежность (доверительная вероятность) оценки** $\Theta^*$ - вероятность $\gamma$, для которой выполняется неравенство:

$\gamma = \mathbb{P}\{ \delta > |\Theta - \Theta^*| \} = \mathbb{P}\{\Theta^* - \delta < \Theta < \Theta^* + \delta \}$,

где $\left( \Theta^* - \delta;\Theta^* + \delta \right)$ - **доверительный интервал**.

Доверительный интервал бывает односторонним и двусторонним:

$ \mathbb{P}\{ \beta_1(\gamma) < \Theta < \beta_2(\gamma) \} = p$

$\mathbb{P}\{ \beta(\gamma) < \Theta \} = \gamma$ или $ \mathbb{P}\{ \beta(\gamma) > \Theta \} = \gamma$

**$p = 1-\gamma$ - уровень значимости**

**Нулевая гипотиза $H_0$** - выдвинутая (основная) гипотеза.
**Конкурирующая гипотеза $H_1$** - альтарнативная гипотеза, противоречащая $H_0$

**Ошибка первого рода** - отвергается верная нулевая гипотеза.

**Ошибка второго рода** - принимается неверная нулевая гипотеза.

**Уровень значимости $\alpha$** - вероятность совешить ошибку первого рода.

**$\beta$** - вероятность ошибки второго рода.

**Мощность критерия $1 - \beta$** - мощность критерия.

Для проверки стат. гипотез используются **критерии**, с помощью которых вычисляются **наблюденные** и **критические** значения случайных величин этих критериев. На основании этих значений выделяют **критическую область** и **область принятия нулевой гипотезы**.





## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $\chi^2 $. Дать содержательную интерпретацию полученным результатам.

### Пункт 1.
Вычислить точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности $\gamma \in \{0.95, 0.99\}$. Сделать выводы.

In [15]:
from google.colab import files

uploaded = files.upload()

import pandas as pd
df = pd.read_csv("team33_pr1.csv")
df

Saving team33_pr1.csv to team33_pr1 (1).csv


,PS,PPY
0,123.274162,2694.000000
1,93.688363,3448.333333
2,114.398422,2213.571429
3,147.928994,3375.000000
4,123.274162,2830.833333
...,...,...
106,148.000000,3419.750000
107,93.688363,3525.000000
108,114.398422,2500.000000
109,123.274162,2358.000000


In [16]:
import numpy as np
from scipy.stats import t, chi, norm, chi2

x_s = 115.94
N = 111
S = 14.78

In [17]:
df = N - 1
for gamma in [0.95, 0.99]:
    t_gamma = t.ppf((1+gamma)/2, df)

    left_bound = x_s - t_gamma*S/np.sqrt(N)
    right_bound = x_s + t_gamma*S/np.sqrt(N)

    print(f"γ = {gamma}")
    print(f"Доверительный интервал: ({round(left_bound,4)}; {round(right_bound,4)})")
    print(f"Точность Δ = {round(t_gamma*S/np.sqrt(N),4)}\n")

γ = 0.95
Доверительный интервал: (113.1599; 118.7201)
Точность Δ = 2.7801

γ = 0.99
Доверительный интервал: (112.2627; 119.6173)
Точность Δ = 3.6773



По использованным из предыдущей практической работы значению выборочного среднего, выборочного СКО, объему выборки и вычисленной квантили $t_{\gamma}$ при уровне надежности $\gamma = 0.95$ был вычислен доверительный интервал для математического ожидания исследуемой случайной величины и точность интервала.
В результате вычислений получен доверительный интервал для математического ожидания $(113.1599;118.7201)$ и точность интервала ${Δ}=2.7801$.

Используется распределение Стьюдента, поскольку генеральное среднеквадратичное отклонение неизвестно и заменяется выборочной оценкой (S).

Доверительный интервал представляется как: $$\left(\bar{x}_в - \frac{t_{\gamma}S}{\sqrt{N}}; \bar{x}_в + \frac{t_{\gamma}S}{\sqrt{N}}\right),$$

где $t_{\gamma}$ удовлетворяет условию:

$$\mathbb{P}\left(\left|\frac{\bar{x}_в - a}{S/\sqrt{N}}\right| < t_{\gamma}\right) = 2\int_0^{t_{\gamma}}s(t,N)dt = \gamma = 2S(t_\gamma,N) - 1$$

$$S(t_\gamma,N) = \frac{\gamma}{2}+\frac{1}{2}$$

отсюда можно найти квантиль:

$$t_\gamma = S^{-1} \left( \frac{1+\gamma}{2},N \right)$$

При уровне надежности $\gamma = 0.99$ доверительный интервал для математического ожидания принимает вид $(112.2627;119.6173)$, а точность интервала составляет ${Δ}=3.6773$.

При увеличении уровня надёжности доверительный интервал расширяется, что соответствует теоретическим положениям математической статистики.

### Пункт 2.
Для вычисления границ доверительного интервала для среднеквадратичного отклонения определить значение $q$ при заданных $\gamma$ и $n$. Построить доверительные интервалы, сделать выводы.

In [18]:
# табличные значения q
q_values = {
    0.95: 0.147,
    0.99: 0.196
}

for gamma in [0.95, 0.99]:
    q = q_values[gamma]

    left_bound = S * (1 - q)
    right_bound = S * (1 + q)

    print(f"γ = {gamma}")
    print(f"q = {q}")
    print(f"Доверительный интервал СКО: ({round(left_bound,4)}; {round(right_bound,4)})\n")

γ = 0.95
q = 0.147
Доверительный интервал СКО: (12.6073; 16.9527)

γ = 0.99
q = 0.196
Доверительный интервал СКО: (11.8831; 17.6769)



  Для вычисления доверительного интервала СКО воспользуемся соотношением $\mathbb{P} = \left( |S-\sigma|<\delta \right) = \gamma$.

  Преобразуем:

  $$ S-\delta < \sigma < S+\delta$$

  Чтобы определить надежность $\gamma$ проведем некоторые преобразования.
  При замене $q=\frac{\delta}{S}$ его можно представить как:

  $$ S(1-q) < \sigma < S(1+q),$$

  где $q<1$


  $$\frac{1}{S(1+q)}<\frac{1}{\sigma}<\frac{1}{S(1-q)}$$



  $$\frac{\sqrt{N-1}}{S(1+q)}<\frac{\sqrt{N-1}}{\sigma}<\frac{\sqrt{N-1}}{S(1-q)}$$

  Далее воспользуемся $\chi = \frac{S\sqrt{N-1}}{\sigma}$ и преобразуем интервал:

  $$\frac{\sqrt{N-1}}{(1+q)}<\chi<\frac{\sqrt{N-1}}{(1-q)}$$

  Чтобы вероятность выполнения неравенства равнялась заданной надежности, должно выполняться условие:

  $$\int_{\frac{\sqrt{N-1}}{(1+q)}}^{\frac{\sqrt{N-1}}{(1-q)}}R(\chi,N)d\chi=\gamma$$

  Значение $q$ было найдено с помощью таблицы по заданным значениям $\gamma$ и $N$. Далее доверительный интервал был посчитан с помощью третьего двойного неравнства.



### Пункт 3.
Проверить гипотезу о нормальности заданного распределения с помощью критерия $\chi^2$ (Пирсона). Для этого необходимо найти теоретические частоты и вычислить наблюдаемое значение критерия. Для удобства вычисления необходимо заполнить табл. 1.

In [19]:
x = np.array([88.73, 96.66, 104.5585, 112.457, 120.3555, 128.254, 136.1525, 144.051])
m = np.array([0.009009, 0.180180, 0.099099, 0.261261, 0.279279, 0.000000, 0.036036, 0.135135])
n = np.array([1, 20, 11, 29, 31, 0, 4, 15])
h = 7.90

intervals = np.array([
    [84.749,  92.711],
    [92.711, 100.609],
    [100.609, 108.508],
    [108.508, 116.406],
    [116.406, 124.305],
    [124.305, 132.203],
    [132.203, 140.102],
    [140.102, 148.000]
], dtype=float)

# перерасчет интервалов
intervals[0] = np.array([-np.inf, intervals[0, 1]])
intervals[-1] = np.array([intervals[-1, 0], np.inf])

for interval in intervals:
    interval[0] = (interval[0] - x_s) / S
    interval[1] = (interval[1] - x_s) / S

# вычисление вероятностей попадания СВ в интервалы
p = np.zeros(len(intervals))
for index, interval in enumerate(intervals):
    p[index] = norm.cdf(interval[1]) - norm.cdf(interval[0])

# вычисление значений теоретических частот
n_t = np.zeros(len(intervals))
for i, p_i in enumerate(p):
    n_t[i] = N * p_i

print("p =", np.round(p, 4))
print("n_t =", np.round(n_t, 4))


p = [0.058  0.0918 0.1577 0.205  0.2017 0.1501 0.0845 0.051 ]
n_t = [ 6.4398 10.1883 17.5086 22.7593 22.3905 16.6628  9.3844  5.6663]


Таблица 1

| i | $[x_i, x_{i+1})$ | $n_i$ | $p_i$ | $n_i'$ | $\left(n_i-n_i' \right)^2$ | $\frac{\left( n_i-n_i' \right) ^2}{n_i'}$ | $n_i^2$ | $\frac{n_i^2}{n_i'}$ |
|---|---|---|---|---|---|---|---|---|
| 1 | [$-\infty$, -1.5720) | 1 | 0.0580 | 6.4398 | 29.5911 | 4.5948 | 1 | 0.1553 |
| 2 | [-1.5720, -1.0373) | 20 | 0.0918 | 10.1883 | 96.2692 | 9.4490 | 400 | 39.2614 |
| 3 | [-1.0373, -0.5028) | 11 | 0.1577 | 17.5086 | 42.3619 | 2.4195 | 121 | 6.9100 |
| 4 | [-0.5028, 0.0315) | 29 | 0.2050 | 22.7593 | 38.9468 | 1.7112 | 841 | 36.9517 |
| 5 | [0.0315, 0.5658) | 31 | 0.2017 | 22.3905 | 74.1242 | 3.3105 | 961 | 42.9196 |
| 6 | [0.5658, 1.1003) | 0 | 0.1501 | 16.6628 | 277.6508 | 16.6628 | 0 | 0.0000 |
| 7 | [1.1003, 1.6348) | 4 | 0.0845 | 9.3844 | 28.9928 | 3.0893 | 16 | 1.7040 |
| 8 | [1.6348, $+\infty$) | 15 | 0.0510 | 5.6663 | 87.1194 | 15.3750 | 225 | 39.2210 |
|   | $\sum$ | 111 | 1.0000 | 111.0000 | - | **56.6120** | - | **167.6120** |

### Пункт 4.
Доказать, что
$$
\chi^2_{\text{набл}} = \sum_{i=1}^k \frac{n_i^2}{n_i'} - n.
$$

Покажем эквивалентность этой формулы основной формуле критерия Пирсона:

$$
\chi^2_{\text{набл}} = \sum_{i=1}^k \frac{(n_i - n_i')^2}{n_i'}.
$$

Раскроем квадрат в числителе:

$$
\chi^2_{\text{набл}} =
\sum_{i=1}^k \frac{n_i^2 - 2n_i n_i' + n_i'^2}{n_i'} =
\sum_{i=1}^k \frac{n_i^2}{n_i'} - 2\sum_{i=1}^k \frac{n_i n_i'}{n_i'} + \sum_{i=1}^k \frac{n_i'^2}{n_i'}.
$$

Упростим выражение:

$$
\chi^2_{\text{набл}} =
\sum_{i=1}^k \frac{n_i^2}{n_i'} - 2\sum_{i=1}^k n_i + \sum_{i=1}^k n_i'.
$$

Так как сумма наблюдаемых частот равна объёму выборки, и сумма теоретических частот также равна объёму выборки, то

$$
\sum_{i=1}^k n_i = n, \qquad \sum_{i=1}^k n_i' = n.
$$

Следовательно,

$$
\chi^2_{\text{набл}} =
\sum_{i=1}^k \frac{n_i^2}{n_i'} - 2n + n =
\sum_{i=1}^k \frac{n_i^2}{n_i'} - n.
$$

Таким образом, доказано, что

$$
\chi^2_{\text{набл}} = \sum_{i=1}^k \frac{n_i^2}{n_i'} - n.
$$

Проконтролируем корректность вычисления $\chi^2_{\text{набл}}$ по обеим формулам.

In [20]:
# Проверка эквивалентности формул для χ²

# χ² по основной формуле Пирсона
chi_1 = np.sum((n - n_t)**2 / n_t)

# χ² по преобразованной формуле
chi_2 = np.sum(n**2 / n_t) - N

print(f"χ²_набл по основной формуле: {round(chi_1, 4)}")
print(f"χ²_набл по преобразованной формуле: {round(chi_2, 4)}")

if np.isclose(chi_1, chi_2):
    print("Вычисления верны!")
else:
    print("Есть расхождение в вычислениях.")

χ²_набл по основной формуле: 56.612
χ²_набл по преобразованной формуле: 56.612
Вычисления верны!


### Пункт 5.
По заданному уровню значимости $\alpha = 0.05$ и числу степеней свободы $df$ найти критическую точку $\chi^2_{крит}$ и сравнить с наблюдаемым значением. Сделать выводы.

In [21]:
K = len(x) # число интервалов
alpha = 0.05
df_crit = K - 3 # число степеней свободы
chi2_crit = chi2.ppf(1 - alpha, df=df_crit)
chi2_observ = np.sum((n - n_t)**2 / n_t)
print("Число степеней свободы:", df_crit)
print("Критическое:", round(chi2_crit, 4))
print("Наблюдаемое:", round(chi2_observ, 4))
if chi2_observ <= chi2_crit:
  print("Не отвергаем гипотезу H0")
else:
  print("Отвергаем гипотезу H0")

Число степеней свободы: 5
Критическое: 11.0705
Наблюдаемое: 56.612
Отвергаем гипотезу H0


Были рассчитаны значения $χ^2_{набл}$ и $χ^2_{крит}$ при уровне значимости $α= 0.05$. Число степеней свободы: $$ df = K - 3 = 8 - 3 = 5. $$ Критическое значение критерия Пирсона: $$ \chi^2_{\text{крит}} = 11.0705 $$ Наблюдаемое значение критерия: $$ \chi^2_{\text{набл}} = 56.612 $$ Так как $$ \chi^2_{\text{набл}} > \chi^2_{\text{крит}}, $$ то нулевая гипотеза $H_0$ о нормальном распределении выборки отвергается. Следовательно, при уровне значимости $\alpha = 0.05$ выборка не согласуется с нормальным распределением.

Проведем аналогичные операции со второй выборкой.

In [22]:

x = np.array([1990.243, 2478.229, 2964.271, 3450.3125, 3936.354, 4422.396, 4908.4375, 5394.479])
n = np.array([0.117117, 0.315315, 0.279279, 0.081081, 0.117117, 0.045045, 0.018018, 0.027027])
abs_freq = np.array([13, 35, 31, 9, 13, 5, 2, 3])
h = 486.04
intervals = np.array([
    [1745.278, 2235.208],
    [2235.208, 2721.25],
    [2721.25, 3207.292],
    [3207.292, 3693.333],
    [3693.333, 4179.375],
    [4179.375, 4665.417],
    [4665.417, 5151.458],
    [5151.458, 5637.5]
], dtype=float)

N = np.sum(abs_freq)

# Выборочное среднее и выборочное СКО по сгруппированным данным
x_s = np.sum(x * abs_freq) / N
S = np.sqrt(np.sum(abs_freq * (x - x_s) ** 2) / (N - 1))

print("Исходные расчётные характеристики:")
print(f"N = {N}")
print(f"x̄ = {round(x_s, 4)}")
print(f"S = {round(S, 4)}")

print("\n1. Доверительный интервал для математического ожидания")
df = N - 1

for gamma in [0.95, 0.99]:
    t_gamma = t.ppf((1 + gamma) / 2, df)
    delta = t_gamma * S / np.sqrt(N)
    left_bound = x_s - delta
    right_bound = x_s + delta

    print(f"\nγ = {gamma}")
    print(f"Доверительный интервал мат. ожидания: ({round(left_bound, 4)}; {round(right_bound, 4)})")
    print(f"Точность Δ = {round(delta, 4)}")

print("\n2. Доверительный интервал для СКО")

q_values = {
    0.95: 0.147,
    0.99: 0.196
}

for gamma in [0.95, 0.99]:
    q = q_values[gamma]
    left_bound = S * (1 - q)
    right_bound = S * (1 + q)

    print(f"\nγ = {gamma}")
    print(f"q = {q}")
    print(f"Доверительный интервал СКО: ({round(left_bound, 4)}; {round(right_bound, 4)})")

print("\n3. Проверка гипотезы о нормальности по критерию Пирсона")

std_intervals = intervals.copy()
std_intervals[0, 0] = -np.inf
std_intervals[-1, 1] = np.inf

for interval in std_intervals:
    interval[0] = (interval[0] - x_s) / S
    interval[1] = (interval[1] - x_s) / S

p = np.zeros(len(std_intervals))
for i, interval in enumerate(std_intervals):
    p[i] = norm.cdf(interval[1]) - norm.cdf(interval[0])

n_t = np.zeros(len(std_intervals))
for i, p_i in enumerate(p):
    n_t[i] = N * p_i

chi_terms = (abs_freq - n_t) ** 2 / n_t
n_sq = abs_freq ** 2
n_sq_div_nt = n_sq / n_t

chi2_observ = np.sum((abs_freq - n_t) ** 2 / n_t)
print(f"\nНаблюдаемое значение χ²: {round(chi2_observ, 4)}")

print("\n4. Контроль вычислений χ²")

chi_1 = np.sum((abs_freq - n_t) ** 2 / n_t)
chi_2 = np.sum(abs_freq ** 2 / n_t) - N

print(f"χ²_набл по основной формуле: {round(chi_1, 4)}")
print(f"χ²_набл по преобразованной формуле: {round(chi_2, 4)}")

if np.isclose(chi_1, chi_2):
    print("Вычисления верны!")
else:
    print("Есть расхождение в вычислениях.")

print("\n5. Сравнение χ²_набл и χ²_крит")

K = len(x)
alpha = 0.05
df_crit = K - 3

chi2_crit = chi2.ppf(1 - alpha, df=df_crit)

print(f"Число степеней свободы: {df_crit}")
print(f"Критическое: {round(chi2_crit, 4)}")
print(f"Наблюдаемое: {round(chi2_observ, 4)}")

if chi2_observ <= chi2_crit:
    print("Не отвергаем гипотезу H0")
else:
    print("Отвергаем гипотезу H0")


Исходные расчётные характеристики:
N = 111
x̄ = 3016.5881
S = 801.2282

1. Доверительный интервал для математического ожидания

γ = 0.95
Доверительный интервал мат. ожидания: (2865.8764; 3167.2998)
Точность Δ = 150.7117

γ = 0.99
Доверительный интервал мат. ожидания: (2817.243; 3215.9332)
Точность Δ = 199.3451

2. Доверительный интервал для СКО

γ = 0.95
q = 0.147
Доверительный интервал СКО: (683.4477; 919.0087)

γ = 0.99
q = 0.196
Доверительный интервал СКО: (644.1875; 958.2689)

3. Проверка гипотезы о нормальности по критерию Пирсона

Наблюдаемое значение χ²: 35.404

4. Контроль вычислений χ²
χ²_набл по основной формуле: 35.404
χ²_набл по преобразованной формуле: 35.404
Вычисления верны!

5. Сравнение χ²_набл и χ²_крит
Число степеней свободы: 5
Критическое: 11.0705
Наблюдаемое: 35.404
Отвергаем гипотезу H0


Таблица 2

| i | $[x_i, x_{i+1})$ | $n_i$ | $p_i$ | $n_i'$ | $\left(n_i-n_i' \right)^2$ | $\frac{\left( n_i-n_i' \right) ^2}{n_i'}$ | $n_i^2$ | $\frac{n_i^2}{n_i'}$ |
|---|---|---|---|---|---|---|---|---|
| 1 | [$-\infty$, -0.9752) | 13 | 0.1647 | 18.2843 | 27.9240 | 1.5272 | 169 | 9.2429 |
| 2 | [-0.9752, -0.3686) | 35 | 0.1915 | 21.2550 | 188.9240 | 8.8884 | 1225 | 57.6334 |
| 3 | [-0.3686, 0.2380) | 31 | 0.2379 | 26.4019 | 21.1429 | 0.8008 | 961 | 36.3989 |
| 4 | [0.2380, 0.8446) | 9 | 0.2068 | 22.9523 | 194.6666 | 8.4814 | 81 | 3.5291 |
| 5 | [0.8446, 1.4513) | 13 | 0.1258 | 13.9641 | 0.9296 | 0.0666 | 169 | 12.1024 |
| 6 | [1.4513, 2.0579) | 5 | 0.0536 | 5.9444 | 0.8919 | 0.1500 | 25 | 4.2056 |
| 7 | [2.0579, 2.6645) | 2 | 0.0159 | 1.7700 | 0.0529 | 0.0299 | 4 | 2.2599 |
| 8 | [2.6645, $+\infty$) | 3 | 0.0039 | 0.4279 | 6.6156 | 15.4596 | 9 | 21.0317 |
|   | $\sum$ | 111 | 1.0000 | 111.0000 | - | **35.4040** | - | **146.4040** |


По данной выборке были вычислены доверительные интервалы для математического ожидания и среднеквадратичного отклонения, а также проверена гипотеза о нормальности распределения с помощью критерия Пирсона.

Наблюдаемое значение критерия составило $χ^2_{набл} = 35.4040$, критическое значение при уровне значимости $\alpha = 0.05$ и числе степеней свободы $df = 5$ равно $χ^2_{крит} = 11.0705$.

Так как $χ^2_{набл} > χ^2_{крит}$, нулевая гипотеза H0 о нормальном распределении отвергается. Следовательно, данная выборка не согласуется с нормальным распределением.


## Выводы

В ходе выполнения практической работы были получены практические навыки нахождения интервальных статистических оценок параметров распределения выборочных данных, а именно математического ожидания и среднеквадратичного отклонения.

Для первой выборки была получена интервальная оценка математического ожидания генеральной совокупности с уровнем надежности 95%:
$a \in (113.1599; 118.7201)$.

Для второй выборки интервальная оценка математического ожидания при уровне надежности 95% имеет вид:
$a \in (2865.8764; 3167.2998)$.

Также были получены интервальные оценки среднеквадратичного отклонения генеральной совокупности при уровне надежности 95%:
$\sigma \in (12.6073; 16.9527)$ — для первой выборки,
$\sigma \in (683.4477; 919.0087)$ — для второй выборки.

Кроме того, была проверена гипотеза согласия исследуемых распределений с нормальным законом с использованием критерия Пирсона $\chi^2$ при уровне значимости $\alpha = 0.05$.

По результатам расчетов наблюдаемое значение статистики критерия оказалось больше критического значения как для первой, так и для второй выборки. В связи с этим нулевая гипотеза о нормальном распределении в обоих случаях была отвергнута.

Таким образом, можно сделать вывод, что исследуемые выборочные данные не согласуются с нормальным распределением.